Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/TicketService.hs

```diff
@@ -594,3 +690,282 @@ postTicketBookingsUpdateSeats _ TicketBookingUpdateSeatsReq {..} = do
               }
       QTSM.create seatM
   pure Kernel.Types.APISuccess.Success
+
+postTicketServiceCancel :: (Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person), Kernel.Types.Id.Id Domain.Types.Merchant.Merchant) -> API.Types.UI.TicketService.TicketServiceCancelReq -> Environment.Flow Kernel.Types.APISuccess.APISuccess
+postTicketServiceCancel (_mbPersonId, merchantId) req = do
+  now <- getCurrentTime
+  bHoursRes <- QBH.findById req.businessHourId >>= fromMaybeM (BusinessHourNotFound req.businessHourId.getId)
+  fork "cancel and refund:" $ do
+    isLockAcquired <- isUserTicketCancellationLock req.businessHourId req.date req.ticketServiceCategoryId
```

Comment:
Use `tryLockRedis` inside this as done in `tryInitTriggerLock` instead of having a lock separately on line 701 otherwise this will still be open to race condition